agent/service_manager.go

```diff
@@ -335,6 +338,7 @@ func makeConfigRequest(bd BaseDeps, addReq AddServiceRequest) *structs.ServiceCo
 		Name:           name,
 		Datacenter:     bd.RuntimeConfig.Datacenter,
 		QueryOptions:   structs.QueryOptions{Token: addReq.token},
+		MeshGateway:    ns.Proxy.MeshGateway,
```

Comment:
Before GA we should figure out if these changes on both sides of the wire for this RPC strongly implies an upgrade order when upgrading to 1.10. I think it's probably correct to do servers before clients given how this was updated, but it's worth writing that down.

Reply:
Our standard upgrade process already recommends upgrading servers before clients:
https://www.consul.io/docs/upgrading#standard-upgrades

Do you mean something more strongly worded?

Reply:
Right. Sorry I was mixing up the servers/clients and primary-dc/secondary-dc upgrade order. The former is specified and the latter is not.